src/IO/Observer/ReductionActions.hpp

```diff
@@ -679,5 +679,50 @@ struct WriteReductionData {
     }
   }
 };
+
+/*!
+ * \brief Write a single row of data to the reductions file without the need to
+ * register or reduce anything, e.g. from a singleton component or from a
+ * specific chare.
+ *
+ * Use observers::Actions::ContributeReductionData instead if you need to
+ * perform a reduction before writing to the file.
+ *
+ * Invoke this action on the observers::ObserverWriter component on node 0. Pass
+ * the following arguments when invoking this action:
+ *
+ * - `subfile_name`: the name of the `h5::Dat` subfile in the HDF5 file. Include
+ *   a leading slash, e.g., `/element_data`.
+ * - `legend`: a `std::vector<std::string>` of column labels for the quantities
+ *   being observed (e.g. `{"Time", "L1ErrorDensity", "L2ErrorDensity"}`).
+ * - `reduction_data`: a `std::tuple<...>` with the data to write. The tuple can
+ *   hold either `double`s or `std::vector<double>`s and is flattened before it
+ *   is written to the file to form a single row of values. The total number of
+ *   values must match the length of the `legend`.
+ */
+struct WriteSingletonReductionData {
```

Comment:
I'm not a fan of the name `WriteSingletonReductionData` as this doesn't have to be run on singletons (according to your description above), and because this is no longer a true reduction. Maybe something like `WriteSingleRowOfData` or `WriteSingleRowToSubfile`? (it doesn't have to be one of these, but that's the best I could come up with at the moment 😛 )

Reply:
I went with `WriteReductionDataRow`, keeping the "ReductionData" in the name to make clear it writes to the reductions file, but removing the "singleton" from the name.